util/llbutil/authprovider/cloudauth/authprovider.go

```diff
@@ -287,27 +340,38 @@ func (ap *authProvider) getAuthConfig(ctx context.Context, host string) (*authCo
 
 	// check user's secrets
 	ac, err := ap.getAuthConfigForProject(ctx, "", "", host)
-	if err == nil {
+	if err != nil {
+		if !errors.Is(err, ErrNoCredentialsFound) {
+			return nil, fmt.Errorf("failed to lookup credentials for %s: %w", host, err)
+		}
+	} else {
 		ap.authConfigCache[host] = ac
 		return ac, nil
 	}
-	if err != ErrNoCredentialsFound {
-		ap.console.Warnf("failed to lookup username/password for %s under user secrets: %s", host, err.Error())
-	}
 
 	// fall back to project's secrets (starting with the root-level Earthfile's org/project)
 	for _, op := range ap.orgProjects {
+		exists, err := ap.projectExists(ctx, op.org, op.project)
+		if err != nil {
+			return nil, err
+		}
+		if !exists {
+			ap.console.Warnf("Warning: PROJECT %s/%s was defined; however it doesn't exist (or you don't have access)", op.org, op.project)
+			continue
+		}
+
 		ac, err := ap.getAuthConfigForProject(ctx, op.org, op.project, host)
-		if err == nil {
+		if err != nil {
+			if !errors.Is(err, ErrNoCredentialsFound) {
+				return nil, fmt.Errorf("failed to lookup credentials for %s: %w", host, err)
+			}
+		} else {
 			ap.authConfigCache[host] = ac
 			return ac, nil
 		}
```

Comment:
Similar comment here wrt it being unclear how the `if` branch flows back to the main logic:
```suggestion
		if err == nil {
			ap.authConfigCache[host] = ac
			return ac, nil
		}
		if !errors.Is(err, ErrNoCredentialsFound) {
			return nil, fmt.Errorf("failed to lookup credentials for %s: %w", host, err)
		}
```